static/app/views/integrationPipeline/githubInstallationSelect.tsx

```diff
@@ -0,0 +1,141 @@
+import {Fragment, useState} from 'react';
+import styled from '@emotion/styled';
+import * as qs from 'query-string';
+
+import {addLoadingMessage} from 'sentry/actionCreators/indicator';
+import {BaseAvatar} from 'sentry/components/core/avatar/baseAvatar';
+import {Button} from 'sentry/components/core/button';
+import type {SelectKey, SelectOption} from 'sentry/components/core/compactSelect';
+import {CompactSelect} from 'sentry/components/core/compactSelect';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+
+type Installation = {
+  avatar_url: string;
+  github_organization: string;
+  installation_id: number;
+};
+
+type GithubInstallationProps = {
+  installation_info: Installation[];
+};
+
+export function GithubInstallationSelect({installation_info}: GithubInstallationProps) {
+  const [installationID, setInstallationID] = useState<SelectKey>(-1);
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+
+  const handleSubmit = (e: React.MouseEvent, id?: SelectKey) => {
+    e.preventDefault();
+    addLoadingMessage(t('Submitting'));
+    setIsSaving(true);
+
+    const {
+      location: {origin},
+    } = window;
+
+    // redirect to the extensions endpoint with the form fields as query params
+    // this is needed so we don't restart the pipeline loading from the original
+    // OrganizationIntegrationSetupView route
+    const chosen_installation_id = id ?? installationID ?? -1;
+    const currentParams = new URLSearchParams(window.location.search);
+    const newParams = {
+      ...Object.fromEntries(currentParams),
+      chosen_installation_id,
+    };
+
+    const newUrl = `${origin}/extensions/github/setup/?${qs.stringify(newParams)}`;
+    return window.location.assign(newUrl);
+  };
+
+  const handleSelect = ({value}: SelectOption<SelectKey>) => {
+    setInstallationID(value);
+  };
+
+  const selectOptions = installation_info.map(
+    (installation): SelectOption<SelectKey> => ({
+      value: installation.installation_id,
+      label: (
+        <OptionLabelWrapper>
+          <StyledAvatar
+            type="upload"
+            uploadUrl={installation.avatar_url}
+            size={16}
+            title={installation.github_organization}
+          />
+          <span>{`${installation.github_organization}`}</span>
+        </OptionLabelWrapper>
+      ),
+    })
+  );
+
+  return (
+    <Fragment>
+      <StyledContainer>
+        <StyledHeader>{t('Install on an Existing Github Organization')}</StyledHeader>
+        <p>
+          {t(
+            'We noticed you already integrated with Github! Do you want to connect an existing Github organization to this Sentry organization or connect a new one?'
```

Comment:
i think the text should be different if they don't have an existing installation. like we can check `installation_info.length` to see if we even need to render the dropdown at all. otherwise this will be confusing if they're installing it for the first time on any of their accounts

Reply:
or are we skipping this page completely if they're installing it for the first time?

Reply:
ok we will skip it